File: user/src/main/java/com/fx/user/adapter/in/web/dto/UserSignUpRequest.java
package com.fx.user.adapter.in.web.dto;

import com.fx.user.application.port.in.dto.UserSignUpCommand;

public record UserSignUpRequest(
    String email,
    String password,
    String phone,
    String nickname
) {
    public UserSignUpCommand toCommand(UserSignUpRequest signUpRequest) {
        return new UserSignUpCommand(signUpRequest.email, signUpRequest.password, signUpRequest.phone, signUpRequest.nickname);
    }
}

File: user/src/main/kotlin/com/fx/user/adapter/in/web/dto/UserSignUpRequest.kt
package com.fx.user.adapter.`in`.web.dto

import com.fx.user.application.`in`.dto.UserSignUpCommand

data class UserSignUpRequest(
    val email: String,
    val password: String,
    val phone: String,
    val nickname: String
) {
    fun toCommand(): UserSignUpCommand {
        return UserSignUpCommand(
            email = this.email,
            password = this.password,
            phone = this.phone,
            nickname = this.nickname
        )
    }
}